internal/cli/server_upgrade.go

```diff
@@ -84,12 +84,26 @@ func (c *ServerUpgradeCommand) Run(args []string) int {
 	// Error handling from input
 
 	if !c.confirm {
-		c.ui.Output(confirmReqMsg, terminal.WithErrorStyle())
+		proceed, err := c.ui.Input(&terminal.Input{
+			Prompt: "Would you like to proceed with the Waypoint server upgrade? Only 'yes' will be accepted to approve. ",
+			Style:  "",
+			Secret: false,
+		})
+		if err != nil {
+			c.ui.Output(
+				"Error upgrading server: %s",
+				clierrors.Humanize(err),
+				terminal.WithErrorStyle(),
+			)
+		} else if proceed != "yes" {
```

Comment:
We could normalize the input with a `strings.ToLower()` so we can accept any variation of "yes".